pkgs/tools/misc/viddy/default.nix

```
@@ -1,30 +1,39 @@
-{ lib, buildGoModule, fetchFromGitHub }:
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+}:
 
-buildGoModule rec {
+rustPlatform.buildRustPackage rec {
   pname = "viddy";
-  version = "0.4.0";
+  version = "1.0.0";
 
   src = fetchFromGitHub {
     owner = "sachaos";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-iF5b5e3HPT3GJLRDxz9wN1U5rO9Ey51Cpw4p2zjffTI=";
+    hash = "sha256-tI4JyYjgj86mw4DCLkRPTIrofoFZiKkVKtb6b5iak5A=";
   };
 
-  vendorHash = "sha256-/lx2D2FIByRnK/097M4SQKRlmqtPTvbFo1dwbThJ5Fs=";
+  cargoHash = "sha256-A/jVMCsRdb+DR7qebtHmexxej7O6T9Y+cAqg212r9iI=";
 
-  ldflags = [
-    "-s"
-    "-w"
-    "-X"
-    "main.version=${version}"
-  ];
+  # requires nightly features
+  env.RUSTC_BOOTSTRAP = 1;
+
+  env.VERGEN_BUILD_DATE = "2024-08-22"; # managed via the update script
```

```suggestion
  env.VERGEN_BUILD_DATE = "1970-01-01";
```
the date of build is always mocked because otherwise the package is not reproducible.

---

yazi does this https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/ya/yazi-unwrapped/package.nix#L27 does this, cc @eljamm

---

Yazi uses `VERGEN_BUILD_DATE` to for [printing the build version](https://github.com/search?q=repo%3Asxyazi%2Fyazi%20VERGEN_BUILD_DATE&type=code) when invoking `--version` and it does not affect reproducibility.

Checking [viddy's source code](https://github.com/search?q=repo%3Asachaos%2Fviddy%20VERGEN_BUILD_DATE&type=code), this seems to be used in a similar manner, so I don't think it's an issue here.

---

But that means we need to update it with every update and possible build it into a custom update script. IMO not worth the time and effort and we can just mock it with a value to make our lives easy and things straight forward.

---

Personally, I don't consider this to be a maintenance burden. Not only does the update script already take care of this, but it's trivial to get the build date if I were to update the package manually.

So, if this doesn't hurt reproducibility and the package maintainers are willing to keep it updated, I don't see why not.

That said, we should consider how this mocked value might cause confusion for users and upstream maintainers and how this change can improve bug reporting.

---

I agree. It's worth the effort. And already supported by nixpkgs-update. https://github.com/nix-community/nixpkgs-update/issues/182